projects/angular-split/src/lib/gutter/split-gutter-exclude-from-drag.directive.ts

```diff
@@ -1,23 +1,21 @@
-import { Directive, OnInit, OnDestroy, Inject, ElementRef } from '@angular/core'
+import { Directive, OnDestroy, ElementRef, inject } from '@angular/core'
 import { SplitGutterDirective } from './split-gutter.directive'
 import { GUTTER_NUM_TOKEN } from './gutter-num-token'
 
 @Directive({
   selector: '[asSplitGutterExcludeFromDrag]',
   standalone: true,
 })
-export class SplitGutterExcludeFromDragDirective implements OnInit, OnDestroy {
-  constructor(
-    @Inject(GUTTER_NUM_TOKEN) private gutterNum: number,
-    private elementRef: ElementRef<HTMLElement>,
-    private gutterDir: SplitGutterDirective,
-  ) {}
+export class SplitGutterExcludeFromDragDirective implements OnDestroy {
+  private readonly gutterNum = inject(GUTTER_NUM_TOKEN)
+  private readonly elementRef = inject<ElementRef<HTMLElement>>(ElementRef)
+  private readonly gutterDir = inject(SplitGutterDirective)
 
-  ngOnInit(): void {
-    this.gutterDir.addToMap(this.gutterDir.gutterToExcludeDragElementMap, this.gutterNum, this.elementRef)
+  constructor() {
+    this.gutterDir._addToMap(this.gutterDir._gutterToExcludeDragElementMap, this.gutterNum, this.elementRef)
   }
 
   ngOnDestroy(): void {
-    this.gutterDir.removedFromMap(this.gutterDir.gutterToExcludeDragElementMap, this.gutterNum, this.elementRef)
+    this.gutterDir._removedFromMap(this.gutterDir._gutterToExcludeDragElementMap, this.gutterNum, this.elementRef)
   }
```

Comment:
Would it be possible to use the [DestroyRef](https://angular.dev/api/core/DestroyRef) for this part?

```ts
const destroyRef = [inject](https://angular.dev/api/core/inject)(DestroyRef);
// register a destroy callback
const unregisterFn = destroyRef.onDestroy(() => doSomethingOnDestroy());
```